packages/at_secondary_server/test/delete_verb_test.dart

```diff
@@ -96,4 +104,50 @@ void main() {
       expect(paramsMap[AT_KEY], 'privatekey:at_secret');
     });
   });
+
+  group('verify deletion of protected keys', () {
+    var handler =
+        DeleteVerbHandler(mockKeyStore, StatsNotificationService.getInstance());
+
+    test('verify deletion of signing public key', () {
+      var command = 'delete:$AT_SIGNING_PUBLIC_KEY@alice';
+      var paramsMap = getVerbParam(Delete().syntax(), command);
+
+      expect(
+          () =>
+              handler.processVerb(Response(), paramsMap, mockInboundConnection),
+          throwsA(
+              predicate((exception) => exception is UnAuthorizedException)));
+    });
+
+    test('verify deletion of signing private key', () {
+      var command = 'delete:@alice:$AT_SIGNING_PRIVATE_KEY@alice';
+      var paramsMap = getVerbParam(Delete().syntax(), command);
+      expect(
+          () =>
+              handler.processVerb(Response(), paramsMap, mockInboundConnection),
+          throwsA(
+              predicate((exception) => exception is UnAuthorizedException)));
+    });
+
+    test('verify deletion of encryption public key', () {
+      var command = 'delete:$AT_ENCRYPTION_PUBLIC_KEY@alice';
+      var paramsMap = getVerbParam(Delete().syntax(), command);
+      expect(
+          () =>
+              handler.processVerb(Response(), paramsMap, mockInboundConnection),
+          throwsA(
+              predicate((exception) => exception is UnAuthorizedException)));
+    });
+
```

Comment:
You need to add one more test asserting that the pkam public key may not be deleted

Reply:
This has been added as part of 73086dfc94ab73767bd16b3968ade74f3da04b0c